headers.h

```diff
@@ -11,15 +11,30 @@
 #define HAVE_SETSID
 
 /* OS-specific includes */
-#if defined(__GLIBC__)
+#if defined(__GLIBC__) || defined(__CYGWIN__)
+# define HAVE_UNISTD_H
+# define HAVE_SYS_IOCTL_H
 # define HAVE_SYS_SYSMACROS_H
 # define HAVE_ERROR_H
+# define HAVE_TTYNAME
 #endif
 
 /* Now we actually include crap ;) */
+#ifdef HAVE_UNISTD_H
+# include <unistd.h>
+#endif
 #ifdef HAVE_ERROR_H
 # include <error.h>
 #endif
+#ifdef HAVE_SYS_IOCTL_H
+# include <sys/ioctl.h>
+#endif
 #ifdef HAVE_SYS_SYSMACROS_H
 # include <sys/sysmacros.h>
 #endif
+
+#if defined(__GNUC_STDC_INLINE__) && (__GNUC_STDC_INLINE__ - 0)
```

Comment:
No Clang anywhere here: It should "work", but may not inline anything eventually.